Terradue/OWSLib
# Improve Authentication Options (#580)

* add username/password/cert/verify for WMS classes

* add WMTS cert/verify capability

* add username/password/cert/verify to WCS classes

* add username/password/cert/verify to WFS classes

* wmts/tms/wcs/wfs/wms should now support username/password/cert/verify

* returns 11 collections, verified using browser

new collection since test was written is "Dutch addresses (subset
Otterlo). OGR GeoPackage Driver" at
https://demo.pygeoapi.io/master/collections/ogr_addresses_gpkg

test now passes

* updated layer to one that exists

neither layer in the test exists on this GeoServer
using a new layer that does exist
also added checks for Exception in the string to be clearer about
expected results

* update version to 0.17.2

* default should be verify=True

* versions back to previous until formal release

* add Authentication class for local/global auth usage

change calls to openURL or requests.get/post to use the auth instance.
this automatically attacheds username/password/cert/verify to requests.

* pass auth instance to openURL, only overwrite verify if not set

* remove extra import, resolve conflict for test

* typo cookie -> cookies

* use auth in new api() function

function was added in bd0eded6488dc98ff35226ec094d7d2920e0bc39

* pass Authentication instance to functions instead of using auth.openURL
